### PR TITLE
added options for events that ignore custom-event-name-casing

### PR DESCRIPTION
**Pull request by devTeaa**
Hi, currently there are some components from library that still uses pascalCase when passing the function
`<customComponent @maskClick="handleMaskClick"/>` this changes added options so it can ignore some event naming just like in `component-name-in-template-casing`.

What do you think?

### CODE REVIEW

**Review by ota-meshi — CHANGES_REQUESTED** [review]
Thank you for this PR! 

Perhaps this PR solves https://github.com/vuejs/eslint-plugin-vue/issues/1260#issuecomment-662104360.
Could you add test cases to check ignores `input:update`, `search:update` and `click:row`?

Could you also add description of the new option to the documentation?
https://github.com/vuejs/eslint-plugin-vue/blob/master/docs/rules/custom-event-name-casing.md

**Comment by devTeaa** [comment]
Updated the docs

image

File: tests/lib/rules/custom-event-name-casing.js [hunk 1]
```diff
@@ -166,6 +166,66 @@ tester.run('custom-event-name-casing', rule, {
...
+          onClick() {
+            this.$emit('input:update')
+            this.$emit('search:update')
+            this.$emit('click:row')
+          }
+        }
+      }
+      </script>
+      `,
+      options: [{ ignores: ['input:update', 'search:update', 'click:row'] }]
```
Comment (ota-meshi) [inline tests/lib/rules/custom-event-name-casing.js:228]: Could you replace it with a testcase that uses regular expressions?

```suggestion
      options: [{ ignores: ['/^[a-z]+(?:-[a-z]+)*:[a-z]+(?:-[a-z]+)*$/u'] }]
```

File: docs/rules/custom-event-name-casing.md [hunk 2]
```diff
@@ -49,7 +49,16 @@ export default {
...
-Nothing.
+
+```json
+{
+  "vue/custom-event-name-casing": ["error", {
+    "ignores": []
+  }]
+}
+```
+- `ignores` (`string[]`) ... The event names to ignore. Sets the event name to allow. For example, custom event names, Vue components event with special name, or Vue library component event name. You can set the regexp by writing it like `"/^name/"` or `update:input` or `fooBar`.
```
Comment (ota-meshi) [inline docs/rules/custom-event-name-casing.md:60]: `update:*` is already ignored without using the `ignores` option, so I think it's easier for people to understand by using the `click:row` in Vuetify.

```suggestion
- `ignores` (`string[]`) ... The event names to ignore. Sets the event name to allow. For example, custom event names, Vue components event with special name, or Vue library component event name. You can set the regexp by writing it like `"/^name/"` or `click:row` or `fooBar`.
```

File: docs/rules/custom-event-name-casing.md [hunk 3]
```diff
@@ -49,7 +49,16 @@ export default {
...
+
```
Comment (ota-meshi) [inline docs/rules/custom-event-name-casing.md:61]: Could you add an example to the documentation that actually uses the option?



````md
### `"ignores": ["fooBar", "/^[a-z]+(?:-[a-z]+)*:[a-z]+(?:-[a-z]+)*$/u"]`

<eslint-code-block :rules="{'vue/custom-event-name-casing': ['error', {ignores:'/^[a-z]+(?:-[a-z]+)*:[a-z]+(?:-[a-z]+)*$/u'}]}">

```vue
<template>
  <!-- ✓ GOOD -->
  <button @click="$emit('click:row')" />
  <button @click="$emit('fooBar')" />

  <!-- ✗ BAD -->
  <button @click="$emit('myEvent')" />
</template>
<script>
export default {
  methods: {
    onClick () {
      /* ✓ GOOD */
      this.$emit('click:row')
      this.$emit('fooBar')

      /* ✗ BAD */
      this.$emit('myEvent')
    }
  }
}
</script>
```

</eslint-code-block>
````


Other example: https://github.com/vuejs/eslint-plugin-vue/blob/master/docs/rules/no-useless-v-bind.md#ignoreincludescomment-true

File: tests/lib/rules/custom-event-name-casing.js (hunk 1 above)
Comment (ota-meshi) [inline tests/lib/rules/custom-event-name-casing.js:228]: Could you also add testcases that reports a violation that doesn't match the regular expressions.

**Review by ota-meshi — CHANGES_REQUESTED** [review]

File: docs/rules/custom-event-name-casing.md (hunk 2 above)
Comment (devTeaa) [reply docs/rules/custom-event-name-casing.md:60]: This is accidentally the same I swear 😂

**Comment by devTeaa** [comment]
Added an example for `ignores` options

image

File: package.json [hunk 4]
```diff
@@ -5,7 +5,7 @@
   "main": "lib/index.js",
   "scripts": {
     "start": "npm run test:base -- --watch --growl",
-    "test:base": "mocha \"tests/lib/**/*.js\" --reporter dot",
+    "test:base": "mocha \"tests/lib/rules/custom-event-name-casing.js\" --reporter dot",
```
Comment (ota-meshi) [inline package.json:8]: Would you please revert this change?

Comment (devTeaa) [reply package.json:8]: sorry, I will

**Review by devTeaa — COMMENTED** [review]
Done

**Review by ota-meshi — APPROVED** [review]
LGTM! Thanks for your contribution!
I will include this change in the next release.

**Comment by devTeaa** [comment]
Thanks for helping the review!
